winsup/utils/getprocaddr.c

```diff
@@ -82,65 +124,66 @@ main (int argc, char **argv)
 {
   char *end;
 
-  if (argc < 2)
+  if (argc < 4)
     {
-      output (1, "Need a function name\n");
+    output (1, "Need a function name, exit code and pid\n");
       return 1;
     }
 
   if (strcmp(argv[1], "CtrlRoutine"))
     {
-      if (argc > 2)
+    if (argc > 4)
         {
-          output (1, "Unhandled option: %s\n", argv[2]);
+     output (1, "Unhandled option: %s\n", argv[4]);
           return 1;
         }
 
-      HINSTANCE kernel32 = GetModuleHandle ("kernel32");
+    HINSTANCE kernel32 = GetModuleHandle ("kernel32");
       if (!kernel32)
         return 1;
-      void *address = (void *) GetProcAddress (kernel32, argv[1]);
+    void *address = (void *) GetProcAddress (kernel32, argv[1]);
       if (!address)
         return 1;
-      output (0, "%p\n", address);
+    output (0, "%p\n", address);
       return 0;
     }
+  exit_code = atoi(argv[2]);
+  pid = atoi(argv[3]);
 
   /* Special-case kernel32!CtrlRoutine */
-  if (argc == 3 && !strcmp ("--alloc-console", argv[2]))
+  if (argc == 5 && !strcmp ("--alloc-console", argv[4]))
```

Comment:
Umm... why?

I do not actually see any code where this is used, and have to assume that I only needed this for testing purposes... Therefore, this entire `Special-case CtrlRoutine` clause should be dropped.

Reply:
I will remove it.

Reply:
Done https://github.com/msys2/msys2-runtime/pull/31/commits/a2da9e9a73dfb229ce2e9402c1b6c8f5645e768d